src/wallet/wallet_list.h

```diff
@@ -41,4 +41,33 @@
  */
 uint8_t get_wallet_list(const char *wallet_list[]);
 
+/**
+ * @brief This API searches for wallet on the flash using wallet_id as key
+ * @details If a VALID_WALLET is found in the flash, then this API fills the
+ * wallet_name, wallet_info, wallet_id fields of the Wallet structure
+ *
+ * @param wallet_id The wallet_id that needs to be searched
+ * @param wallet Reference to Wallet structure which will be populated by the
+ * function
+ * @return true If the wallet corresponding to wallet_id is found in the flash
+ * and is in VALID_WALLET state.
+ * @return false If the wallet corresponding to wallet_id is not found or is not
+ * in VALID_WALLET state.
+ */
+bool wallet_selector(const uint8_t *wallet_id, Wallet *wallet);
+
+/**
+ * @brief This API searches for wallet on the flash using wallet_id as key
+ * @details If a VALID_WALLET is found in the flash, then this API fills the
+ * wallet_name buffer if provided
+ *
+ * @param wallet_id The wallet_id that needs to be searched
+ * @param wallet_name The buffer in which wallet name will be filled or NULL if
+ * wallet name is not required
+ * @return true If the wallet corresponding to wallet_id is found in the flash
+ * and is in VALID_WALLET state.
+ * @return false If the wallet corresponding to wallet_id is not found or is not
+ * in VALID_WALLET state.
+ */
+bool get_wallet_data(const uint8_t *wallet_id, uint8_t *wallet_name);
```

Comment:
The name is confusing to me, we could rename as `find_valid_wallet`.

This function is not being used right now, What would be the intended usage of this function

Reply:
This function is for the coin application to get the name of the wallet.. Name of the wallet is required to trigger user confirmation before any operation on the wallet

Reply:
I have updated the name to `get_wallet_name_by_id`